gtsfm/runner/gtsfm_runner_base.py

```diff
@@ -272,10 +282,14 @@ def setup_ssh_cluster_with_retries(self) -> SSHCluster:
             )
         return cluster
 
-    def run(self) -> GtsfmData:
+    def run(self, graph_partitioner: GraphPartitionerBase = None) -> GtsfmData:
         """Run the SceneOptimizer."""
         start_time = time.time()
 
+        # Create graph partitioner if not provided
+        if graph_partitioner is None:
+            graph_partitioner = SinglePartition()
```

Comment:
remove the input argument and just intiialize it here, add a TODO to create a config. :)